game/src/sandbox/dashboards/commuter.rs

```diff
@@ -30,11 +30,20 @@ pub struct CommuterPatterns {
     draw_all_blocks: Drawable,
 }
 
-struct CurrentBlock {
-    id: BlockID,
-    draw: Drawable,
-    // After clicking a block, show its outline to indicate it's been clicked.
-    sticky: Option<Drawable>,
+#[derive(PartialEq, Clone, Copy)]
+enum BlockSelection {
+    NothingSelected,
+    Unlocked(BlockID),
+    Locked {
```

Comment:
I initially had separate LockedWithComparison and LockedWithoutComparison cases as in your example but I feel the code ended up cleaner this way. 

See c5796f990c43927e1167d9df3f5b21cc6a9ead38 for where I consolidated them.

Reply:
I like this better! The fully expanded enum didn't feel quite right, nor did the compressed struct with tightly coupled fields